addons/mail/static/src/models/message_list_view/message_list_view.js

```diff
@@ -6,11 +6,89 @@ import { attr, one } from '@mail/model/model_field';
 registerModel({
     name: 'MessageListView',
     identifyingFields: ['threadViewOwner'],
+    recordMethods: {
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsAtBottom() {
+            return (
+                this.scrollTop >= this.scrollHeight - this.clientHeight - this.endThreshold
+            );
+        },
+        /***
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsAtEnd() {
+            if (this.threadViewOwner.order === 'asc') {
+                return this.isAtBottom;
+            }
+            return this.isAtTop;
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsAtStart() {
+            if (this.threadViewOwner.order === 'asc') {
+                return this.isAtTop;
+            }
+            return this.isAtBottom;
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsAtTop() {
+            return this.scrollTop <= this.endThreshold;
+        },
+    },
     fields: {
+        clientHeight: attr(),
         /**
          * States the OWL component of this message list view
          */
         component: attr(),
+        /**
+         * The margin that we use to detect that the scrollbar is a the end of
+         * the threadView.
+         */
+        endThreshold: attr({
+            default: 30,
+        }),
+        /**
+         * States whether the message list scroll position is at the bottom
+         * of the message list.
+         */
+        isAtBottom: attr({
+            compute: '_computeIsAtBottom',
+        }),
+        /**
+         * States whether the message list scroll position is at the end of
+         * the message list. Depending of the message list order, this could be
+         * the top or the bottom.
+         */
+        isAtEnd: attr({
+            compute: '_computeIsAtEnd',
+        }),
+        /**
+         * States whether the message list scroll position is at the start
+         * of the message list. Depending of the message list order, this could
+         * be the top or the bottom.
+         */
+        isAtStart: attr({
+            compute: '_computeIsAtStart',
+        }),
+        /**
+         * States whether the message list scroll position is at the top of
+         * the message list.
+         */
+        isAtTop: attr({
+            compute: '_computeIsAtTop',
+        }),
```

Comment:
Both `isAtTop` and `isAtBottom` are only used as indirection to compute `isAtEnd`, which can be done in a single line as was done in the previous code, so I have doubts about the need to introduce these fields in this PR.

Reply:
The goal was to make the old spaghetti code more readable. I find it easier to read now than some math done on `clientHeight` and `scrollHeight`.